b2sdk/api.py

```diff
@@ -299,11 +301,14 @@ def delete_bucket(self, bucket):
         :type bucket: b2sdk.bucket.Bucket
         """
         account_id = self.account_info.get_account_id()
-        response = self.session.list_buckets(account_id)
-        buckets = BucketFactory.from_api_response(self, response)
-        for bucket in buckets:
-            self.cache.save_bucket(bucket)
-        return self.session.delete_bucket(account_id, bucket.id_)
+        # trying to find the selected bucket in cache. If not found, resetting the bucket cache!
```

Comment:
In place of performing cache update every time, this will update if the bucket is not found in cache